### PR TITLE
Update nodes.md

### PR DESCRIPTION
**Pull request by navanshu**
Fixed the incorrect typescript in the code examples

### CODE REVIEW

**Review by zurfyx — CHANGES_REQUESTED** [review]
Thank you, that's Flow typing, these examples were never updated. If you want to go ahead with these corrections you probably want to type `null | undefined | LexicalNode` instead

See an example
https://github.com/facebook/Lexical//blob/d05ae76e5ab452ef8ee977f74ed4e890ac7c5db0/packages/lexical/src/nodes/LexicalTextNode.ts#L1125-L1129

**Comment by navanshu** [comment]
> Thank you, that's Flow typing, these examples were never updated. If you want to go ahead with these corrections you probably want to type `null | undefined | LexicalNode` instead
> 
> See an example https://github.com/facebook/Lexical//blob/d05ae76e5ab452ef8ee977f74ed4e890ac7c5db0/packages/lexical/src/nodes/LexicalTextNode.ts#L1125-L1129

Yeah, did think of that, just started getting into lexical.

File: packages/lexical-website/docs/concepts/nodes.md [hunk 1]
```diff
@@ -183,7 +183,7 @@ export function $createCustomParagraphNode(): ParagraphNode {
   return new CustomParagraph();
 }
 
-export function $isCustomParagraphNode(node: ?LexicalNode): boolean {
+export function $isCustomParagraphNode(node: LexicalNode | null | undefined): boolean {
```
Comment (zurfyx) [inline packages/lexical-website/docs/concepts/nodes.md:186]: Mind updating the return type too on the 3 of them please?
```suggestion
export function $isCustomParagraphNode(node: LexicalNode | null | undefined): node is CustomParagraph  {
```

**Review by zurfyx — APPROVED** [review]

File: packages/lexical-website/docs/concepts/nodes.md (hunk 1 above)
Comment (navanshu) [reply packages/lexical-website/docs/concepts/nodes.md:186]: Instance of will always return a boolean, I think the return type is correct, it is upto typescript to the type of Node

Comment (zurfyx) [reply packages/lexical-website/docs/concepts/nodes.md:186]: > Instance of will always return a boolean, I think the return type is correct, it is upto typescript to assert the type of Node

While `boolean` is correct, TS can't infer that the type is CustomParagraph unless it's specifically marked like it, which can come useful in many ocassions. See predicates examples - https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

Comment (navanshu) [reply packages/lexical-website/docs/concepts/nodes.md:186]: :D :O
This is interesting, I had no idea about this.

**Comment by zurfyx** [comment]
LGTM, thank you!

**Pull request merged**